utils/dist_build.sh

```diff
@@ -0,0 +1,130 @@
+
+echo "Clearing CLI temp folder"
+rm -rf $cliTmpDir
+mkdir -p $cliTmpDir
+
+echo "Preparing CLI Build"
+cd $ORGDIR
+[ -d $cliDIR ] || git clone https://github.com/TykTechnologies/tyk-cli.git
+cd $cliDIR
+git checkout master
+git pull
+go get -v ./...
+gox -osarch="linux/arm64 linux/amd64 linux/386"
+
+echo "Copying CLI Build files"
+mv tyk-cli_linux_* $cliTmpDir/
+
+echo "Starting Tyk build"
+cd $SOURCEBINPATH
+
+echo "Blitzing TGZ dirs"
+for arch in ${!ARCHTGZDIRS[@]}
+do
+    rm -rf ${ARCHTGZDIRS[$arch]}
+    mkdir -p ${ARCHTGZDIRS[$arch]}
+done
+
+echo "Building Tyk binaries"
+gox -osarch="linux/arm64 linux/amd64 linux/386" -tags 'coprocess grpc'
+
+echo "Building Tyk CP binaries"
+export CPBINNAME_LUA=tyk_linux_amd64_lua
+export CPBINNAME_PYTHON=tyk_linux_amd64_python
+
+gox -osarch="linux/amd64" -tags 'coprocess python' -output '{{.Dir}}_{{.OS}}_{{.Arch}}_python'
+gox -osarch="linux/amd64" -tags 'coprocess lua' -output '{{.Dir}}_{{.OS}}_{{.Arch}}_lua'
+
+TEMPLATEDIR=${ARCHTGZDIRS[i386]}
+echo "Prepping TGZ Dirs"
+mkdir -p $TEMPLATEDIR/apps
+mkdir -p $TEMPLATEDIR/js
+mkdir -p $TEMPLATEDIR/middleware
+mkdir -p $TEMPLATEDIR/middleware/python
+mkdir -p $TEMPLATEDIR/middleware/lua
+mkdir -p $TEMPLATEDIR/event_handlers
+mkdir -p $TEMPLATEDIR/event_handlers/sample
+mkdir -p $TEMPLATEDIR/templates
+mkdir -p $TEMPLATEDIR/policies
+mkdir -p $TEMPLATEDIR/utils
+mkdir -p $TEMPLATEDIR/install
+
+cp $SOURCEBINPATH/apps/app_sample.json $TEMPLATEDIR/apps
+cp $SOURCEBINPATH/templates/*.json $TEMPLATEDIR/templates
+cp -R $SOURCEBINPATH/install/* $TEMPLATEDIR/install
+cp $SOURCEBINPATH/middleware/*.js $TEMPLATEDIR/middleware
+cp $SOURCEBINPATH/event_handlers/sample/*.js $TEMPLATEDIR/event_handlers/sample
+cp $SOURCEBINPATH/js/*.js $TEMPLATEDIR/js
+cp $SOURCEBINPATH/policies/*.json $TEMPLATEDIR/policies
+cp $SOURCEBINPATH/tyk.conf.example $TEMPLATEDIR/
+cp $SOURCEBINPATH/tyk.conf.example $TEMPLATEDIR/tyk.conf
+cp -R $SOURCEBINPATH/coprocess $TEMPLATEDIR/
+
+# Clone template dir to all architectures and copy corresponding binaries
+for arch in ${!ARCHTGZDIRS[@]}
+do
+    archDir=${ARCHTGZDIRS[$arch]}
+    [ $archDir != $TEMPLATEDIR ] && cp -R $TEMPLATEDIR/* $archDir
+    mv tyk_linux_${arch/i386/386} $archDir/$SOURCEBIN
+    cp $cliTmpDir/tyk-cli_linux_${arch/i386/386} $archDir/utils/$CLIBIN
+done
+mv $CPBINNAME_LUA ${ARCHTGZDIRS[amd64]}/$SOURCEBIN-lua
+mv $CPBINNAME_PYTHON ${ARCHTGZDIRS[amd64]}/$SOURCEBIN-python
+
+echo "Compressing"
+for arch in ${!ARCHTGZDIRS[@]}
+do
+    cd ${ARCHTGZDIRS[$arch]}/../
+    tar -pczf ${ARCHTGZDIRS[$arch]}/../tyk-linux-$arch-$VERSION.tar.gz tyk.linux.$arch-$VERSION/
+done
+
+if [ $BUILDPKGS == "1" ]; then
+    CONFIGFILES=(--config-files /opt/tyk-gateway/apps --config-files /opt/tyk-gateway/templates --config-files /opt/tyk-gateway/middleware --config-files /opt/tyk-gateway/event_handlers --config-files /opt/tyk-gateway/js --config-files /opt/tyk-gateway/policies --config-files /opt/tyk-gateway/tyk.conf)
```

Comment:
This will make for some really complex diffs - perhaps do one `--config-files foo` per line?

Also, given that the flag name is in plural, could you not specify the flag only once at the start, or use something like a comma-separated list? (just wondering, hoping that you don't have to repeat it over and over)